src/main/scala/mesosphere/marathon/core/task/tracker/impl/TaskOpProcessorImpl.scala

```diff
@@ -64,51 +77,96 @@ private[tracker] object TaskOpProcessorImpl {
+    stateChange.flatMap {
+      case stateChange: TaskStateChange.Expunge =>
         // Used for task termination or as a result from a UpdateStatus action.
         // The expunge is propagated to the taskTracker which in turn informs the sender about the success (see Ack).
         repo.expunge(op.taskId.idString).map { _ =>
-          taskTrackerRef ! TaskTrackerActor.TaskRemoved(op.taskId, TaskTrackerActor.Ack(op.sender))
-        }.recoverWith(tryToRecover(op)(expectedTaskState = None))
-
-      case Action.UpdateStatus(status) =>
-        statusUpdateActionResolver.resolve(op.taskId, status).flatMap { action: Action =>
-          // Since this action is mapped to another action, we delegate the responsibility to inform
-          // the sender to that other action.
-          process(op.copy(action = action))
-        }
+          taskTrackerRef ! TaskTrackerActor.TaskRemoved(op.taskId, TaskTrackerActor.Ack(op.sender, stateChange))
+        }.flatMap { _ =>
+          notifyOthers(op.stateOp, stateChange)
+        }.recoverWith(tryToRecover(op)(expectedTaskState = None, oldTaskState = Some(stateChange.task)))
+
+      case stateChange: TaskStateChange.Failure =>
+        // Used if a task status update for a non-existing task is processed.
+        // Since we did not change the task state, we inform the sender directly of the failed operation.
+        op.sender ! Status.Failure(stateChange.cause)
+        Future.successful(())
 
-      case Action.Noop =>
+      case stateChange: TaskStateChange.NoChange =>
         // Used if a task status update does not result in any changes.
         // Since we did not change the task state, we inform the sender directly of the success of
         // the operation.
-        op.sender ! (())
+        op.sender ! stateChange
         Future.successful(())
 
-      case Action.Fail(cause) =>
-        // Used if a task status update for a non-existing task is processed.
-        // Since we did not change the task state, we inform the sender directly of the failed operation.
-        op.sender ! Status.Failure(cause)
+      case stateChange: TaskStateChange.Update =>
+        // Used for a create or as a result from a UpdateStatus action.
+        // The update is propagated to the taskTracker which in turn informs the sender about the success (see Ack).
+        val marathonTask = TaskSerializer.toProto(stateChange.task)
+        repo.store(marathonTask).map { _ =>
+          taskTrackerRef ! TaskTrackerActor.TaskUpdated(stateChange.task, TaskTrackerActor.Ack(op.sender, stateChange))
+        }.flatMap { _ =>
+          notifyOthers(op.stateOp, stateChange)
+        }.recoverWith(tryToRecover(op)(
+          expectedTaskState = Some(stateChange.task), oldTaskState = stateChange.oldTask))
+    }
+  }
+
+  private[this] def notifyOthers(stateOp: TaskStateOp, stateChange: TaskStateChange): Future[Unit] = {
+    // compatibility: if it's a statusUpdate that resulted in a Update or Expunge, do as before
```

Comment:
Am I missing something or are you only propagating mesos status updates?

We also need to propagate other updates -- at least to the launch queue (which will forward them to the `AppTaskLauncherActor`).